sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -265,19 +267,20 @@ public void writeRows(String placementName, Iterator<MigrationScanResult> iterat
         DeltaPlacement placement = (DeltaPlacement) _placementCache.get(placementName);
         Session session = placement.getKeyspace().getCqlSession();
         BatchStatement statement = new BatchStatement(BatchStatement.Type.LOGGED);
-        final Semaphore semaphore = new Semaphore(maxConcurrentWrites);
         AtomicReference<Throwable> error = new AtomicReference<>();
+        RateLimiter rateLimiter = RateLimiter.create(1000);
+        Phaser phaser = new Phaser();
```

Comment:
I think Phaser works for what you're doing.  Just as an alternate, do you think it'd be simpler to just maintain all of the futures from `session.executeAsync(statement)` in list and then at the end just do something like `futures.forEach(Futures::getUnchecked)`?

Reply:
In my opinion, it seems like the phaser is lighter weight, as we won't have to store millions of futures in memory.